### PR TITLE
Update the members list for the recipients of a sharing

### PR DESCRIPTION
**Pull request by nono**

### CODE REVIEW

File: pkg/sharing/member.go [hunk 1]
```diff
@@ -86,6 +88,9 @@ func (s *Sharing) AddContact(inst *instance.Instance, contactID string) error {
 // UpdateRecipients updates the list of recipients
 func (s *Sharing) UpdateRecipients(inst *instance.Instance, members []Member) error {
 	for i, m := range members {
+		if i >= len(s.Members) {
+			s.Members = append(s.Members, Member{})
```
Comment (Gara64) [inline pkg/sharing/member.go]: Don't you want to append `m` instead of `Member{}`?

File: pkg/sharing/sharing.go [hunk 2]
```diff
@@ -196,6 +197,12 @@ func (s *Sharing) CreateRequest(inst *instance.Instance) error {
 	s.UpdatedAt = time.Now()
 	s.Credentials = make([]Credentials, 1)
 
+	for i, m := range s.Members {
+		if contact, err := contacts.FindByEmail(inst, m.Email); err == nil {
```
Comment (Gara64) [inline pkg/sharing/sharing.go]: `err != nil` ?

File: pkg/sharing/member.go [hunk 3]
```diff
@@ -81,6 +81,22 @@ func (s *Sharing) AddContact(inst *instance.Instance, contactID string) error {
 	return nil
 }
 
+// UpdateRecipients updates the list of recipients
+func (s *Sharing) UpdateRecipients(inst *instance.Instance, members []Member) error {
+	for i, m := range members {
+		if m.Email != s.Members[i].Email {
+			if contact, err := contacts.FindByEmail(inst, m.Email); err == nil {
```
Comment (Gara64) [inline pkg/sharing/member.go]: `err != nil` ?

File: pkg/sharing/oauth.go [hunk 4]
```diff
@@ -146,7 +146,7 @@ func CreateOAuthClient(inst *instance.Instance, m *Member) (*oauth.Client, error
 	}
 	cli := oauth.Client{
 		RedirectURIs: []string{m.Instance + "/sharings/answer"},
-		ClientName:   "Sharing " + m.Name,
+		ClientName:   "Sharing " + m.PublicName,
```
[the remark this answers is not in the document]
Comment (nono) [reply pkg/sharing/oauth.go]: In the oauth package, we add a suffix when trying a register a client with an already used client name: https://github.com/cozy/cozy-stack/blob/master/pkg/oauth/client.go#L224-L252

File: pkg/sharing/sharing.go (hunk 2 above)
Comment (nono) [reply pkg/sharing/sharing.go]: No, it's if `err` is nil that the contact is found and valid and that we can use its name. We can safely ignore errors here.

Comment (Gara64) [reply pkg/sharing/sharing.go]: My mistake, I should have reviewed this review

**Pull request merged**